source/fab/tools/tool.py

```diff
@@ -139,7 +139,7 @@ def __str__(self):
 
     def run(self,
             additional_parameters: Optional[
-                Union[str, List[Union[Path, str]]]] = None,
+                Union[str, Sequence[Union[Path, str]]]] = None,
```

Comment:
Not sure what you mean. If I use `List`, mypy complains:
```
tools/tool.py:66: error: Argument 1 to "run" of "Tool" has incompatible type "str | list[str]"; expected "str | list[Path | str] | None"  [arg-type]
```
Line 66 is:
```
            self.run(self._availability_option)
```
with `self._availability_option` getting assigned a value of type:
```
Union[str, List[str]]
```

I think I understand the reason: the `run` method might modify the list and add an element of type `Path` to it, since the argument is declared to include `Path`, but this would then not be compatible with `List[str]` anymore. I assume a generic Sequence does not allow modifications (??) - hence it can't add a variable of a different type into the parent list??

A string might be a sequence, but that would indicate that we would iterate over each character I think

Reply:
I was wondering whether `Sequence` includes `string`, and why you changed from list to sequence. Thanks for the explanation.